Jint/Native/JsDate.cs

```diff
@@ -56,6 +56,6 @@ public override string ToString()
             return "Infinity";
         }
 
-        return ToDateTime().ToString("ddd MMM dd yyyy HH:mm:ss 'GMT'zzz", CultureInfo.InvariantCulture);
+        return _engine.Realm.Intrinsics.Date.PrototypeObject.ToDateString(DateValue).ToString();
```

Comment:
do other JS object follow this pattern?

Reply:
meaning JsNumber, ...

Reply:
well the primitives are quite limited with toString, should we just ignore this - don't you like the behavior?

Reply:
Well JsNumber does follow JS semantincs:

```
    public override string ToString()
    {
        return TypeConverter.ToString(_value);
    }
```

Reply:
Changed to use `TypeConverter` here too